File: C++/Sorting/InsertionSort.cpp
#include <iostream>

using namespace std;

void InsertionSort(int x[], int n)
{
    for (int index = 1; index < n; index++) 
    {
        int holder = x[index];
        int j = index-1;
        while (x[j] > holder && j>=0) 
        {
            x[j+1] = x[j];
            j--;
        }

        j++;

        x[j] = holder;  
    }
}

void PrintArr(int l[], int n)
{
    for(int i=0; i < n; i++)
    {
        cout << l[i] << " ";
    }
    cout << '\n';
}

int main()
{
    cout << "Enter the number of elements: ";
    int n; cin >> n;
    cout << "Enter the elements: ";
    int arr[n];
    for(int i=0; i < n; i++)
    {
        cin >> arr[i];
    }

    InsertionSort(arr, n);

    cout << "Sorted array: ";
    PrintArr(arr, n);
    
    return 0;
}

/*
Input:
    Enter the number of elements: 10
    Enter the elements: 10 9 8 7 6 5 4 3 2 1

Output:
    Sorted array: 1 2 3 4 5 6 7 8 9 10
*/

File: JavaScript/Sorting/InsertionSort.js
function InsertionSort(x)
{
    for (let index = 1; index < x.length; index++) {
        var holder = x[index];
        var j = index-1;
        while (x[j] > holder && j>=0) {
            x[j+1] = x[j];
            j--;
        }
        j++;
        x[j] = holder;  
    }
}

function PrintArray(x)
{
    x.forEach(element => {
        process.stdout.write(element + " ");
    });
    console.log();
}

const n = prompt("Enter the number of elements: ");
const arr = new Array(n);
console.log("Enter the elements: ");
for (let index = 0; index < n; index++) {
    arr[index] = parseInt(prompt(), 10);
}

InsertionSort(arr);
console.log("Sorted array: ");
PrintArray(arr);

/*
Input:
    Enter the number of elements: 10
    Enter the elements: 10 9 8 7 6 5 4 3 2 1

Output:
    Sorted array: 1 2 3 4 5 6 7 8 9 10
*/